### ISSUE TITLE
5.1.2 find_country_by_any_name returns Zimbabwe if name is nil or blank

### ISSUE THREAD

**Issue by nrankin** [countries/countries#776]
ISO3166::Country.find_country_by_any_name('')
ISO3166::Country.find_country_by_any_name(nil)

Both return Zimbabwe which I imagine is the last item in the list.
It should return nil

**Comment by pmor** [comment]
This looks like a duplicate of #746

Are you loading any custom country data, or using Rails 6?

I've checked that 5.1.2 is working correctly both standalone and with Rails 7, but #746 has two reports of issues with Rails 6, including on 5.1.2, that I haven't reproduced yet.

Could you attach your Gemfile, please?

**Comment by nrankin** [comment]
My apologies, it is indeed a duplicate. 
We're on Ruby 2.7.5 and Ruby 2.7.5
Not sure how it could relate, but we don't include activerecord as we're using mongodb.

[gemfile.txt](https://github.com/countries/countries/files/9825926/gemfile.txt)

**Comment by pmor** [comment]
Thanks, I'll try to setup some blank Rails 5 and 6 apps to see if I can reproduce this

**Comment by curtiseinsmann** [comment]
I'm running into this as well. It's happening when `Rails.env` is `"test"`.

```ruby
[1] pry(main)> Rails.env
=> "development"
[2] pry(main)> ISO3166::Country.find_country_by_any_name(nil)&.alpha2
=> nil
[3] pry(main)> ISO3166::Country["ZW"].translated_names
=> ["Zimbabwe",
 "Zimbabwe",
 "Simbabwe",
 "Zimbabue",
 "Zimbabwe",
 "Zimbabwe",
 "Zimbabwe",
 "Zimbabwe",
 "Zimbabwe",
 "Zimbabve",
 "Zimbabwe",
 "Зимбабве",
 "Zimbabwe",
 "Zimbabwe",
 "Zimbabwe",
 "Zimbabwe",
 "Zimbabve",
 "Zimbabwe",
 "Zimbabwe",
 "Zimbabwe",
 "Zimbabve"]
```

```ruby
[1] pry(main)> Rails.env
=> "test"
[2] pry(main)> ISO3166::Country.find_country_by_any_name(nil)&.alpha2
=> "ZW"
[3] pry(main)> ISO3166::Country["ZW"].translated_names
=> ["Zimbabwe",
 "زمبابوي",
 "Зимбабве",
 nil,
 "Zimbabwe",
 nil,
 nil,
 nil,
 "Simbabwe",
 "Zimbabwe",
 nil,
 nil,
 nil,
 nil,
 nil,
 nil,
 nil,
 nil,
 nil,
 nil,
 nil,
 nil,
 nil,
 nil,
 nil,
 nil,
 nil,
 nil,
 "Zimbabue",
 "زیمبابوه",
 nil,
 "Zimbabwe",
 nil,
 nil,
 "זימבבואה",
 "Զիմբաբվե",
 "Zimbabwe",
 "Zimbabwe",
 "ジンバブエ",
 "짐바브웨",
 "Zimbabvė",
 "Zimbabve",
 nil,
 nil,
 "Zimbabwe",
 nil,
 "Zimbabwe",
 "Zimbábue",
 "Zimbabwe",
 "Зимбабве",
 "Zimbabwe",
 "Zimbabwe",
 "ซิมบับเว",
 "Zimbabve",
 "Зімбабве",
 "Xim-ba-bu-ê",
 "津巴布韦",
 "辛巴威",
 "Zimbabwe",
 "Zimbabwe",
 "Zimbabve",
 "Zimbabwe",
 "Zimbabwe",
 "Zimbabwe",
 "Zimbabwe"]
```

Top of my `Gemfile`:

```ruby
source "https://rubygems.org"

ruby "3.0.3"

gem "rails", "7.0.4"
gem "rake", "13.0.6"
```

**Comment by pmor** [comment]
@curtiseinsmann Thanks, that does give me something to hunt for. The app I've been using to diagnose this is configured with a single locale and passes on that configuration to the gem.

Could you check the output of this in `test` vs `development` envs?

`ISO3166.configuration.locales`  and `I18n.available_locales`

Thanks!

**Comment by curtiseinsmann** [comment]
> Could you check the output of this in `test` vs `development` envs?
> 
> `ISO3166.configuration.locales` and `I18n.available_locales`

### test

```ruby
pry(main)> ISO3166.configuration.locales
=> [:en,
 :ar,
 :bg,
 :"ca-CAT",
 :ca,
 :"da-DK",
 :"de-AT",
 :"de-CH",
 :de,
 :ee,
 :"en-AU",
 :"en-BORK",
 :"en-CA",
 :"en-GB",
 :"en-IND",
 :"en-MS",
 :"en-NEP",
 :"en-NG",
 :"en-NZ",
 :"en-PAK",
 :"en-SG",
 :"en-TH",
 :"en-UG",
 :"en-US",
 :"en-ZA",
 :"en-au-ocker",
 :"es-AR",
 :"es-MX",
 :es,
 :fa,
 :"fi-FI",
 :fr,
 :"fr-CA",
 :"fr-CH",
 :he,
 :hy,
 :id,
 :it,
 :ja,
 :ko,
 :lt,
 :lv,
 :"mi-NZ",
 :"nb-NO",
 :nl,
 :"no-NO",
 :pl,
 :"pt-BR",
 :pt,
 :ru,
 :sk,
 :sv,
 :th,
 :tr,
 :uk,
 :vi,
 :"zh-CN",
 :"zh-TW",
 :da,
 :nb,
 :sl,
 :cs,
 :fi,
 :hu,
 :ro]

pry(main)> I18n.available_locales
=> [:en, :ja, :de, :pt, :es, :he, :it, :nl, :ro, :ru, :zh, :fr, :fi, :"pt-BR", :"zh-CN"]
```

### development

```ruby
pry(main)> ISO3166.configuration.locales
=> [:en,
 :da,
 :de,
 :es,
 :fr,
 :it,
 :nb,
 :nl,
 :pt,
 :sl,
 :sv,
 :bg,
 :ca,
 :cs,
 :fi,
 :hu,
 :lv,
 :pl,
 :ro,
 :sk,
 :tr]

pry(main)> I18n.available_locales
=> [:en, :ja, :de, :pt, :es, :he, :it, :nl, :ro, :ru, :zh, :fr, :fi, :"pt-BR", :"zh-CN"]
```

**Comment by pmor** [comment]
I believe this was fixed in  #777
